commands.js

```diff
@@ -1782,8 +1782,17 @@ var commands = exports.commands = {
 
 	addplayer: function (target, room, user) {
 		if (!target) return this.parse('/help addplayer');
-
-		return this.parse('/roomplayer ' + target);
+		target = this.splitTarget(target);
+		var targetUser = this.targetUser;
+		if (!targetUser) return this.sendReply("User " + this.targetUsername + " not found.");
+		if (!room.joinbattle) return this.sendReply("You can only do this in battle rooms.");
```

Comment:
Sorry I wasn't exactly clear, I was only referring to the capitalization, you still need <code>room.</code>. The code is case sensitive, so <code>joinBattle</code> is the name of the function while <code>joinbattle</code> is not defined in the room prototype.